Adds `REAMDE.md` template for all new packages (#1)
pulsar-cooperative/.github: guides/publishing-new-versions.md

# Publishing New Versions

> The process to publish a new package version has attempted to find the middle ground between a hands off approach, and automation. Tending to not lean to heavily into automation.

Reviewer: ```suggestion
The process to publish a new package version is one that tries to find the middle ground between a hands on approach, and automation. Tending to not lean to heavily into automation.
```

I assume that's what you meant? since "hands off" and "automation" are kind of the same thing I believe

Author: Lol, yes this is what I meant. Thank you

> The process to publish a new package version is one that tries to find the middle ground between a hands on approach, and automation. Tending to not lean to heavily into automation.

Reviewer 2: ```suggestion
The process of publishing a new package version is one that tries to find the middle ground between a hands on approach and automation, tending to not lean too heavily into automation.
```

> The general workflow of publishing a new version will look like this.

Reviewer 2: ```suggestion
The general workflow of publishing a new version looks like this.
```

> Keeping in mind that it's assumed at this point, the initial new package has already been published, and the GitHub Secret `PUBLISHED` has been set to `true` (**Match the case!**)

Reviewer 2: ```suggestion
Keeping in mind that it's assumed at this point that the initial new package has already been published, and the GitHub Secret `PUBLISHED` has been set to `true` (**Match the case!**).
```

1. A contributor creates a new PR to the repository. It has been successfully reviewed, and is ready to merge.

> 2. The PR is merged, causing the workflow `version-bump.yml` to trigger, and use Google's `release-please` to draft a new version bump (according to Conventional Commit conventions) and an associated new tag.

Reviewer 2: ```suggestion
2. The PR is merged, causing the workflow `version-bump.yml` to trigger, and use Google's [release-please](https://github.com/googleapis/release-please) to draft a new version bump (according to [Conventional Commit](https://www.conventionalcommits.org/) conventions) and an associated new tag.
```
Just adding some links so people don't need go to and google them.

Author: Great idea here, make people's lives just a tad easier

3. Once this PR is then merged, you are ready to publish a new version!
4. Manually run the workflow `release.yml` to trigger the publication of a new package version to PPR.
5. You're done! As you can see the total process to publish a new version, including the initial contributors PR boils down to:

> * Merge 2 PRs (the original contributors, and the one created by `release-please`)

Reviewer 2: ```suggestion
* Merge 2 PRs (the original contributor's, and the one created by `release-please`)
```

> * Manually trigger 1 workflow, to publish the package.

Reviewer: would it be possible to include multiple content-changing PRs in one version bump? and is it possible to signal something on the release-please PR (by a merge or a comment) to do what manually triggering `release.yml` would do, to make it a bit simpler?

Author: From my understanding, `release-please` makes a new version on every commit. Although I think it is a self canceling one, so ideally if PRs are merged close together it will auto cancel the oldest one.

As for then publishing the version, that'd be amazing, I'm not sure how to do so, but I'll read up on `release-please` to see if it's possible

Author: So initially I was stuck thinking how we could add the publishing a new version to release please, but we don't have to. I'll just add it to the end of that workflow file itself, so it all happens at once

Reviewer: > self canceling one

makes sense if it sees its old PR and closes it before opening a new one, but I'm not familliar on it ><

Author: Wait, I hate to say I'm wrong here.

The version bump PR needs to be merged into the `main` branch, and have the tag published before we can publish the new version, so we cannot do it manually on `release-please` without some sort of plugin? Otherwise we may have to stick with the current method.

Since the other hard part is actions cannot trigger actions, meaning a task can't automatically be created after merging a PR from `release-please`.
